### PR TITLE
Sidebar: Add a sidebar to hold the post settings and inspector

### PR DESCRIPTION
**Pull request by youknowriad**
This PRs triggers the work on the editor's sidebar. I kept it minimal, just opening and closing an empty sidebar when clicking on "Post Settings".

This probably needs more design ❤️ cc @jasmussen

### CODE REVIEW

**Comment by jasmussen** [comment]
Yaaaaay! I love it! I will definitely give it some love and attention one of these days. 

Code looks good too. 

Long term we'll want it shown by default, and its state saved (so if you dismiss it, blog, then come back to the editor, the sidebar is still off until you toggle it back on). We'll want to state-save whether you're using the block editor or text editor too, on that note. Want me to create a separate ticket for this?

**Review by jasmussen — APPROVED** [review]

**Comment by youknowriad** [comment]
Yes, let's create a ticket for that. We may need an API endpoint or use the settings endpoint for these settings.

File: editor/components/button/style.scss [hunk 1]
```diff
@@ -1,8 +1,14 @@
 .editor-button {
 	background: none;
 	border: none;
+	outline: none;
 
 	&:disabled {
 		opacity: 0.6;
 	}
+
+	&.is-active {
```
Comment (aduth) [inline editor/components/button/style.scss]: Do we expect this to be the same styling we'll want for the [`:active` pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:active) ?

If so, should we add it here?
If not, should we be worried about confusion between the two unique states sharing the same name?

File: editor/state.js [hunk 2]
```diff
@@ -148,6 +148,15 @@ export function mode( state = 'visual', action ) {
 	return state;
 }
 
+export function sidebar( state = { opened: false }, action ) {
```
Comment (aduth) [inline editor/state.js]: What other properties do you expect this to hold? Wondering if a simpler `isSidebarOpened` reducer with a boolean value might be more appropriate for now.

File: editor/sidebar/style.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.editor-sidebar {
+	position: fixed;
+	right: 0;
+	top: 32px + $header-height;
```
Comment (aduth) [inline editor/sidebar/style.scss]: We'll need some breakpoint styling for this. Not sure if we wanted to do that here.

File: editor/layout/style.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+.editor-layout.sidebar-opened .editor-layout__content {
+	margin-right: $sidebar-width;
+
+	.editor-text-editor__formatting {
```
Comment (aduth) [inline editor/layout/style.scss]: Does this need to be nested?

File: editor/layout/index.js [hunk 5]
```diff
@@ -2,24 +2,35 @@
...
+import './style.scss';
 import Header from 'header';
+import Sidebar from 'sidebar';
 import TextEditor from 'modes/text-editor';
 import VisualEditor from 'modes/visual-editor';
 
-function Layout( { mode } ) {
+function Layout( { mode, sidebarOpened } ) {
+	const className = classnames( 'editor-layout', {
+		'sidebar-opened': sidebarOpened
```
Comment (aduth) [inline editor/layout/index.js]: As a modifier class, I might expect some boolean-ish prefix to the class name, maybe `has-open-sidebar` ?

File: editor/components/button/style.scss (hunk 1 above)
Comment (youknowriad) [inline editor/components/button/style.scss]: I don't know? :) I've always been confused with the `:active` state. I think `.is-active` and `:active` are different states, maybe we should rename `.is-active`?

File: editor/state.js (hunk 2 above)
Comment (youknowriad) [inline editor/state.js]: I hesitated, I was expecting we may have a property saying we're showing the `PostSettings` or the `BlockInspector`, but yeah. I'll go for `isSidebarOpened` right now!

File: editor/layout/style.scss (hunk 4 above)
Comment (youknowriad) [inline editor/layout/style.scss]: It needs to be nested under `.editor-layout.sidebar-opened`, I can rewrite it :)

File: editor/layout/index.js (hunk 5 above)
Comment (youknowriad) [inline editor/layout/index.js]: or maybe `is-sidebar-opened` to match the reducer name

File: editor/sidebar/style.scss (hunk 3 above)
Comment (youknowriad) [inline editor/sidebar/style.scss]: I'll leave this for a more "designy" eyes :)

File: editor/layout/index.js (hunk 5 above)
Comment (aduth) [inline editor/layout/index.js]: >or maybe `is-sidebar-opened` to match the reducer name

Yep, that works.

**Comment by jasmussen** [comment]
Ticket opened in #450

File: editor/layout/style.scss (hunk 4 above)
Comment (aduth) [inline editor/layout/style.scss]: We could also define this in `modes/text-editor/style.scss` as assuming the existence of the ancestor modifier:

```scss
.editor-text-editor__formatting {
	// ...

	.editor-layout.is-sidebar-opened & {
		// ...
	}
}
```

File: editor/components/button/style.scss (hunk 1 above)
Comment (aduth) [inline editor/components/button/style.scss]: Maybe a question for @jasmussen for prescribing variations on button state we'll need to support.

File: editor/layout/style.scss (hunk 4 above)
Comment (youknowriad) [inline editor/layout/style.scss]: I just learned that we can do `.editor-layout.is-sidebar-opened &` And yes, this is probably better!

File: editor/sidebar/style.scss (hunk 3 above)
Comment (jasmussen) [inline editor/sidebar/style.scss]: I'll definitely tackle this separately, it's already on my to-do 😉

File: editor/state.js (hunk 2 above)
Comment (jasmussen) [inline editor/state.js]: Let's call it PostInspector when the time comes. The idea being that the entire sidebar is the inspector, but when nothing is selected, you are inspecting the post itself, therefore showing the meta boxes.

File: editor/components/button/style.scss (hunk 1 above)
Comment (jasmussen) [inline editor/components/button/style.scss]: I'm thinking about hover, click and active states these days, across all controls, to ensure consistency. Right now the toggled state is visually the same as :active. Did that answer the question?

Comment (youknowriad) [inline editor/components/button/style.scss]: This means all the buttons will have the dark gray background when clicked (before releasing the mouse). I'm not sure we want this, do we?

Comment (jasmussen) [inline editor/components/button/style.scss]: I'm sorry, I'm tired and wasn't being very clear indeed.

My mind has been thinking about the hover state, which right now is blue text and a dark gray outline, which I'm not convinced about. I've been thinking whether we could do the inverted version for both hover, active and toggled. I.e. hover would _also_ have the dark gray background. This is in context of the accessibility and contrast discussions that took place a while back. But we might find that this feels too heavy handed, and that we might be able to have a lighter gray hover+active state, in which case only the toggled state would be dark gray.

Bottomline, probably good to not hover/active/toggled states block this PR from being merged, as this is something I will revisit across all buttons later on regardless.

Comment (youknowriad) [inline editor/components/button/style.scss]: I think I'm renaming `is-active` to `is-toggled` :)

File: editor/components/button/index.js [hunk 6]
```diff
@@ -4,11 +4,12 @@
 import './style.scss';
 import classnames from 'classnames';
 
-function Button( { isPrimary, isLarge, className, ...additionalProps } ) {
+function Button( { isPrimary, isLarge, isActive, className, ...additionalProps } ) {
```
Comment (aduth) [inline editor/components/button/index.js]: Did you intend to rename the prop from `isActive` to `isToggled` as well?

**Review by aduth — APPROVED** [review]
Aside from prop renaming comment, this looks good to me 👍

File: editor/components/button/index.js (hunk 6 above)
Comment (youknowriad) [inline editor/components/button/index.js]: Like I said, "too fast and miss the details"

**Pull request merged**